src/electron/index.ts

```diff
@@ -152,3 +156,17 @@ app.setAsDefaultProtocolClient("ark", process.execPath, ["--"]);
 app.allowRendererProcessReuse = false;
 
 setupPlugins();
+
+const env = new Environment({
+	coins: {
+		ARK,
+	},
+	httpClient: new HttpClient(500),
+	storage: "indexeddb",
+	driver: "memory",
+});
+env.verify()
```

Comment:
Can't use await here (top-level / outside an async function).
What would be the best way to have a single instance of the environment created in the main process so that the handlers can be registered here?

Reply:
Perhaps something like this is enough, isn't it? :hmhm:

Main:

```typescript
import { ipcMain } from "electron";

const env = new Environment({
	coins: {
		ARK,
	},
	httpClient: new HttpClient(500),
	storage: "indexeddb",
	driver: "memory",
});

const initializeEnvironment = async () => {
	await env.verify();
	await env.boot();

    ipcMain.send("env:ready");
}

export const registerEnvironmentHandlers = async () => {
	await initializeEnvironment();
	// setupIpc();
}

```

Renderer:

```jsx
import { ipcRenderer } from "electron";

const App = () => {
	React.useEffect(() => {
		ipcRenderer.on("env:ready", () => {
			hideSplashScreen();
		});
    }, []);

	return <h1>App</h1>;
}
```

Reply:
Let me try. Thanks @luciorubeens